### ISSUE TITLE
Controlled Terminologies / Ontologies / Vocabularies used by BRAPI

### ISSUE THREAD

**Issue by proccaserra** [plantbreeding/API#56]
Is there documentation providing a complete breakdown of the BRAPI attributes that should rely on controlled values?

For location.countryCode/countryName, it seems geoNames must/should be used ( ISO-3166 country code) but what about values for attributes such as 'studyType' ? is it free text?

What about 'seasons' ? Some examples show values such as '2007', which seems to indicate a date/year (ISO8601?) while others state 'spring'. So same question CV or free text?

ObservedVariable object points to 'ontologyDbName' to the variable with 'Variable ontology name (usually a short name)'. Which resources should be used in the context of this attribute? Can you point me to the relevant information?

In short I am looking for information about which semantic resource BRAPI recommends using for the following elements:

-crop
-growthStage
-trait.class
-method.class
-scale.class

**Comment by gcornut** [comment]
A JSON-LD context is being worked on to define all BrAPI JSON entities. It will partially use other ontologies to describe its properties when possible (ex: bioversity MCPD  ontology for germplasm)

**Comment by BrapiCoordinatorSelby** [comment]
If a field in a BrAPI model uses a controlled vocabulary, it will be defined by

- A reference to the particular data standard in the description of that field 
- The field will be defined as an `enum` with a limited set of valid values
- An ontology reference, or list of valid values is defined by the implementing server (ie `Trait` has an ontology reference, there is a GET call to retrieve a list of valid `ObservationLevel` for that server

If a field has none of these things, then it is assumed to be "free text" by the BrAPI standard. That said, each implementing server may (and should) have a more restrictive set of valid options in some cases. Check the API documentation for the servers you are working with to see if they are more restrictive.

> but what about values for attributes such as 'studyType' ? is it free text?

In BrAPI V1, there is a `GET /studytype` request which should return the valid studyTypes for a given server. This may change in V2 if studyType gets adopted to distinguish between "phenotyping", "genotyping", "agronomy", etc.

> What about 'seasons' ?

As of BrAPI 1.2, `season` became an entity which defined a DbId, Year, and Season. The Year is defined as an integer, and the Season is defined as free text ('Spring', 'Q2', 'Season A', etc.). The DbId is sometimes a generic ID and sometimes a concatenation of the other two fields, but should be treated as a DbId in most cases.

 > Which resources should be used in the context of this attribute? Can you point me to the relevant information?

This has changed a bit since the original issue was posted. Some entities, like `ObservationVariable`, have an `ontologyReference`. This object can contain URLs directly to the reference documentation, or can be linked to an `Ontology` entity, defined by the server.

> In short I am looking for information about which semantic resource BRAPI recommends using for the following elements:

All of the fields you listed are defined as "free text" by the current BrAPI standard.
